File: survey-core/src/main/java/org/brinst/surveycore/entity/SurveyOption.java
package org.brinst.surveycore.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@NoArgsConstructor
public class SurveyOption {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	private String option;
	@ManyToOne
	private SurveyQuestion surveyQuestion;

	public SurveyOption(String option) {
		this.option = option;
	}
}

File: survey-data/src/main/kotlin/com/ic/surveydata/entity/SurveyOption.kt
package com.ic.surveydata.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "survey_option")
class SurveyOption {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id = 0

    @Column(nullable = false, unique = true)
    val itemUuid: String = ""

    @Column(nullable = false, unique = false)
    val name: String = ""
}
